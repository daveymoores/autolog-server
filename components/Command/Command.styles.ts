import css from "styled-jsx/css";

export default css`
  header {
    display: flex;
    justify-content: space-between;
    align-items: stretch;
  }

  button {
    width: 50px;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: transparent;
    border: none;
  }

  @media (prefers-color-scheme: dark) {
    button {
      border-left: var(--lineWidth) solid var(--lightGreen);
    }
  }

  @media (prefers-color-scheme: light) {
    button {
      border-left: var(--lineWidth) solid var(--darkGreen);
    }
  }

  h3 {
    text-align: left;
    font-size: 16px;
    font-weight: 500;
    padding: 1em;
    margin: 0;
  }

  code {
    display: block;
    padding: 2em;
    text-align: center;
  }

  @media (prefers-color-scheme: dark) {
    code {
      border-top: var(--lineWidth) solid var(--lightGreen);
      background-color: var(--mistyGrey);
    }
  }

  @media (prefers-color-scheme: light) {
    code {
      border-top: var(--lineWidth) solid var(--darkGreen);
      background-color: var(--mistyGrey);
    }
  }

  section {
    margin-bottom: 50px;
  }

  @media (prefers-color-scheme: dark) {
    section {
      border: var(--lineWidth) solid var(--lightGreen);
    }
  }

  @media (prefers-color-scheme: light) {
    section {
      border: var(--lineWidth) solid var(--darkGreen);
    }
  }
`;
