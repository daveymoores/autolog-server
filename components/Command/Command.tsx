import React from "react";
import useSystemTheme from "react-use-system-theme";

import palette from "../../utils/palette";
import styles from "./Command.styles";

export interface CommandProps {
  code: string;
  heading: string;
}

const Command: React.FC<CommandProps> = ({ code, heading }) => {
  const systemTheme = useSystemTheme("dark");
  return (
    <>
      <section>
        <header>
          <h3>{heading}</h3>
          <button>
            <svg
              width="29px"
              height="29px"
              viewBox="0 0 29 29"
              version="1.1"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink"
            >
              <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
                <g
                  transform="translate(-448.000000, -11.000000)"
                  stroke={
                    systemTheme === "dark"
                      ? palette.LIGHT_GREEN
                      : palette.DARK_GREY
                  }
                  strokeWidth="3"
                >
                  <g transform="translate(448.000000, 11.000000)">
                    <rect x="1.5" y="1.5" width="19" height="19" rx="2" />
                    <rect
                      fill={
                        systemTheme === "dark"
                          ? palette.MISTY_GREY
                          : palette.MISTY_GREEN
                      }
                      x="8.5"
                      y="8.5"
                      width="19"
                      height="19"
                      rx="2"
                    />
                  </g>
                </g>
              </g>
            </svg>
          </button>
        </header>
        <code>{`> ${code}`}</code>
      </section>
      <style jsx>{styles}</style>
    </>
  );
};

export default Command;
