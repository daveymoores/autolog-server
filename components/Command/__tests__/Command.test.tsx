import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";
import React from "react";

import Command, { CommandProps } from "../Command";

// Mock the styles to avoid styled-jsx issues
jest.mock("../Command.styles", () => ({
  __esModule: true,
  default: "",
}));

describe("Command Component", () => {
  const defaultProps: CommandProps = {
    code: "timesheet-gen edit -h 5 -d 8 -m 10 -y 2021",
    heading: "Initialise for current directory",
  };

  it("renders the heading correctly", () => {
    render(<Command {...defaultProps} />);
    expect(screen.getByText(defaultProps.heading)).toBeInTheDocument();
  });

  it("renders code with prefix", () => {
    render(<Command {...defaultProps} />);
    expect(screen.getByText(`> ${defaultProps.code}`)).toBeInTheDocument();
  });

  it("renders SVG button", () => {
    const { container } = render(<Command {...defaultProps} />);
    expect(container.querySelector("svg")).toBeInTheDocument();
  });

  it("renders correctly with empty heading", () => {
    render(<Command code={defaultProps.code} heading="" />);
    const headingElement = screen.queryByRole("heading");
    expect(headingElement?.textContent).toBe("");
  });

  it("has basic component structure", () => {
    const { container } = render(<Command {...defaultProps} />);
    expect(container.querySelector("section")).toBeInTheDocument();
    expect(container.querySelector("header")).toBeInTheDocument();
    expect(container.querySelector("h3")).toBeInTheDocument();
    expect(container.querySelector("code")).toBeInTheDocument();
  });
});
