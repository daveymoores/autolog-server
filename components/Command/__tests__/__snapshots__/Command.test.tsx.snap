// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Command renders Command unchanged 1`] = `
<section
  className="jsx-1811775179"
>
  <header
    className="jsx-1811775179"
  >
    <h3
      className="jsx-1811775179"
    >
      Initialise for current directory
    </h3>
    <button
      className="jsx-1811775179"
    >
      <svg
        className="jsx-1811775179"
        height="29px"
        version="1.1"
        viewBox="0 0 29 29"
        width="29px"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <g
          className="jsx-1811775179"
          fill="none"
          fillRule="evenodd"
          stroke="none"
          strokeWidth="1"
        >
          <g
            className="jsx-1811775179"
            stroke="#e1f7de"
            strokeWidth="3"
            transform="translate(-448.000000, -11.000000)"
          >
            <g
              className="jsx-1811775179"
              transform="translate(448.000000, 11.000000)"
            >
              <rect
                className="jsx-1811775179"
                height="19"
                rx="2"
                width="19"
                x="1.5"
                y="1.5"
              />
              <rect
                className="jsx-1811775179"
                fill="#2d3039"
                height="19"
                rx="2"
                width="19"
                x="8.5"
                y="8.5"
              />
            </g>
          </g>
        </g>
      </svg>
    </button>
  </header>
  <code
    className="jsx-1811775179"
  >
    &gt; timesheet-gen edit -h 5 -d 8 -m 10 -y 2021
  </code>
</section>
`;
